strr-web/stores/strr.ts

```diff
@@ -106,22 +118,46 @@ export const propertyManagerSchema = z.object({
   businessMailingAddress: z.object({
     address: requiredNonEmptyString,
     addressLineTwo: optionalOrEmptyString,
-    city: requiredNonEmptyString,
-    postalCode: requiredNonEmptyString,
+    city: requiredNonEmptyString100,
+    postalCode: requiredNonEmptyString15,
     province: requiredNonEmptyString,
     country: requiredNonEmptyString
   }),
   contact: z.object({
-    firstName: requiredNonEmptyString,
+    firstName: requiredNonEmptyString50,
     middleName: optionalOrEmptyString,
-    lastName: requiredNonEmptyString,
+    lastName: requiredNonEmptyString50,
     preferredName: optionalOrEmptyString,
     phoneNumber: requiredPhone,
     extension: optionalOrEmptyString,
     faxNumber: optionalPhone,
     emailAddress: requiredEmail
   })
-})
+}).refine(
+  data => !data.businessLegalName || data.businessLegalName.length <= 50,
+  {
+    message: 'Business Legal Name must not exceed 50 characters',
+    path: ['businessLegalName']
+  }
+).refine(
+  data => !data.contact.middleName || data.contact.middleName.length <= 50,
+  {
+    message: 'Middle Name must not exceed 50 characters',
+    path: ['contact', 'middleName']
+  }
+).refine(
+  data => !data.contact.preferredName || data.contact.preferredName.length <= 50,
+  {
+    message: 'Preferred Name must not exceed 50 characters',
+    path: ['contact', 'preferredName']
+  }
+).refine(
+  data => !data.contact.extension || data.contact.extension.length <= 15,
+  {
+    message: 'Extension must not exceed 15 characters',
+    path: ['contact', 'extension']
+  }
+)
```

Comment:
`max` doesn't work with `optional` [zod] and this is the way to implement field limit for optional fields.